lib/src/chart/line_chart/line_chart_painter.dart

```diff
@@ -1203,18 +1220,24 @@ class LineChartPainter extends AxisChartPainter<LineChartData>
 
     final Size chartViewSize = getChartUsableDrawSize(viewSize);
 
-    /// Find the nearest spot (on X axis)
-    for (int i = 0; i < barData.spots.length; i++) {
-      final spot = barData.spots[i];
-      if (spot.isNotNull()) {
-        if ((touchedPoint.dx - getPixelX(spot.x, chartViewSize)).abs() <=
-            data.lineTouchData.touchSpotThreshold) {
-          return LineBarSpot(barData, barDataPosition, spot);
-        }
-      }
-    }
+    final closestSpotByX = barData.spots.reduce((lhs, rhs) {
+      final lhsDistance = (touchedPoint.dx - getPixelX(lhs.x, chartViewSize)).abs();
+      final rhsDistance = (touchedPoint.dx - getPixelX(rhs.x, chartViewSize)).abs();
+      return lhsDistance < rhsDistance ? lhs : rhs;
+    });
+
+    final minDistanceX = (touchedPoint.dx - getPixelX(closestSpotByX.x, chartViewSize)).abs();
+
+    final closestSpotByY = barData.spots.where((spot) {
+      final distanceX = (touchedPoint.dx - getPixelX(spot.x, chartViewSize)).abs();
+      return distanceX == minDistanceX;
+    }).reduce((lhs, rhs) {
+      final lhsDistanceY = (touchedPoint.dy - getPixelY(lhs.y, chartViewSize)).abs();
+      final rhsDistanceY = (touchedPoint.dy - getPixelY(rhs.y, chartViewSize)).abs();
+      return lhsDistanceY < rhsDistanceY ? lhs : rhs;
+    });
 
-    return null;
+    return closestSpotByY == null ? null : LineBarSpot(barData, barDataPosition, closestSpotByY);
```

Comment:
I do not feel comfortable by this algorithm,
You even didn't use the `touchSpotThreshold`,
I apologize for this, but It's not gonna be merged.
Please revert all of them, and make a separate PR for this.
This change has a lot of risks, and please first define an issue, then create the PR, and mention the issue, this way we can follow it up.
Thanks!